src/sinks/util/buffer/loki.rs

```diff
@@ -80,6 +120,41 @@ impl Batch for LokiBuffer {
     }
 
     fn push(&mut self, mut item: Self::Input) -> PushResult<Self::Input> {
+        // We must sort the stream labels to ensure they hash to
+        // the same stream if the label set matches.
+        item.labels.sort_unstable();
+
+        let partition = &item.partition;
+        if self.latest_timestamps.is_none() {
+            self.partition = Some(item.partition.clone());
+            self.latest_timestamps = Some(self.global_timestamps.take(&partition));
+        }
+        let latest_timestamp = self
+            .latest_timestamps
+            .as_ref()
+            .unwrap()
+            .get(&item.labels)
+            .cloned()
+            .unwrap_or(item.event.timestamp);
+        if item.event.timestamp < latest_timestamp {
+            match self.out_of_order_action {
+                OutOfOrderAction::Drop => {
+                    warn!(
+                        msg = "Received out-of-order event; dropping event.",
+                        internal_log_rate_secs = 30
+                    );
+                    return PushResult::Ok(false);
+                }
+                OutOfOrderAction::RewriteTimestamp => {
+                    warn!(
+                        msg = "Received out-of-order event, rewriting timestamp.",
+                        internal_log_rate_secs = 30
+                    );
+                    item.event.timestamp = latest_timestamp;
```

Comment:
I realized that it is bad to use `now()` here, because events might be in the past (or a different timezone) and using `now()` will prevent other events from being inserted.